oauthproxy.go

```diff
@@ -725,6 +725,24 @@ func (p *OAuthProxy) SignOut(rw http.ResponseWriter, req *http.Request) {
 		p.ErrorPage(rw, req, http.StatusInternalServerError, err.Error())
 		return
 	}
+
+	session, err := p.getAuthenticatedSession(rw, req)
+	if err == nil && session != nil {
+		providerDatas := p.provider.Data()
+		if providerDatas.BackendLogoutURL != "" {
+			backendLogoutURL := strings.ReplaceAll(providerDatas.BackendLogoutURL, "{id_token}", session.IDToken)
+			resp, err := http.Get(backendLogoutURL) // #nosec G107
+			if err != nil {
+				logger.Errorf("error while calling backend logout: %v", err)
+			} else {
+				defer resp.Body.Close()
+				if resp.StatusCode != 200 {
+					logger.Errorf("error while calling backend logout, %v returned a %v", backendLogoutURL, resp.StatusCode)
+				}
+			}
+		}
+	}
```

Comment:
I think this can be simplified if you refactored this to be in a function. That way you'd be able to use the "return early" pattern. Right now this adds quite a lot of complexity due too the many if-statements.

Reply:
Please reduce the `if` nesting and isolate it into its own function as suggest by @kvanzuijlen.

Something like:

```golang
p.BackendLogout(rw, req)

func (p *OAuthProxy) BackendLogout(rw http.ResponseWriter, req *http.Request) {
	session, err := p.getAuthenticatedSession(rw, req)	

	if err != nil || session == nil {
		// error handling is missing here
		// at least logging
		return;
	}

	providerData := p.provider.Data()
	if providerData.BackendLogoutURL == "" {
		return;
	}

	backendLogoutURL := strings.ReplaceAll(providerData.BackendLogoutURL, "{id_token}", session.IDToken)
	resp, err := http.Get(backendLogoutURL) // #nosec G107

	if err != nil {
		logger.Errorf("error while calling backend logout: %v", err)
	} else {
		defer resp.Body.Close()

		if resp.StatusCode != 200 {
			logger.Errorf("error while calling backend logout, %v returned a %v", backendLogoutURL, resp.StatusCode)
		}
	}
}
```

Reply:
implemented :)